packages/react-native/src/iosUtils.ts

```diff
@@ -0,0 +1,125 @@
+  const framesFromIosCall = framesFromIOSCallStack(data)
+
+  let backtrace = []
+  const backtraceDetails:Record<string, string | Record<string, string | number>[]> = {}
+
+  if (framesFromComponent.length) {
+    backtrace = framesFromComponent
+    backtraceDetails.primaryBackTraceSource = 'ReactNativeComponentStack'
+    if (framesFromReactNativeIos.length) {
+      backtraceDetails.reactNativeIOSStackTrace = framesFromReactNativeIos
+    }
+    if (framesFromIosCall.length) {
+      backtraceDetails.iosCallStack = framesFromIosCall;
+    }
+  } else if (framesFromReactNativeIos.length) {
+    backtrace = framesFromReactNativeIos;
+    backtraceDetails.primaryBackTraceSource = 'ReactNativeIOSStackTrace';
+    if (framesFromIosCall.length) {
+      backtraceDetails.iosCallStack = framesFromIosCall;
+    }
+  } else if (framesFromIosCall.length) {
+    backtrace = framesFromIosCall;
+    backtraceDetails.primaryBackTraceSource = 'iOSCallStack';
+  }
+
+  return { backtrace, backtraceDetails }
+}
+
+function framesFromComponentStack(str:string) {
+  str = str || ''
+  const frames = []
+  const regex = /^\s*in\s(\S+)(\s\(at\s(\S+):(\S+)\)\s*$)?/gm
+  let match
+  while ((match = regex.exec(str)) !== null) {
+    if ( match.index === regex.lastIndex) {
+      regex.lastIndex++;
+    }
+    frames.push({
+      method: match && match.length > 1 ? match[1] : '',
+      file: match && match.length > 3 ? match[3] : '',
+      number: match && match.length > 4 ? match[4] : '',
+    })
+  }
+  return frames
+}
+
+function framesFromReactNativeIosStack(data:NativeExceptionData) {
+  if (!data.reactNativeStackTrace) {
+    return []
+  }
+
+  return data.reactNativeStackTrace.map((frame) => ({
+    method: frame.methodName || '',
+    number: frame.lineNumber || '',
+    file: frame.file || '',
+    column: frame.column || '',
+  }))
+}
+
+function framesFromIOSCallStack(data:NativeExceptionData) {
+  let callStack = []
+
+  if (data.localizedDescription && typeof data.localizedDescription === 'string') {
+    callStack = data.localizedDescription.split('\n').map(item => item.trim())
+  } else if (Array.isArray(data.callStackSymbols) && data.callStackSymbols.length) {
+    callStack = data.callStackSymbols.map(item => item.trim());
+  }
+
+  const frames = [];
+  const regex = /\d+\s+(\S+)\s+(\S+)\s(.+)\s\+\s(\d+)(\s+\((\S+):(\S+)\))?/gm
+  let match
+  callStack.forEach(element => {
+    while ((match = regex.exec(element)) !== null ) {
+      if ( match.index === regex.lastIndex) {
+        regex.lastIndex++;
+      }
+
+      const moduleName = match && match.length > 1 ? match[1] : '';
+      const stackAddress = match && match.length > 2 ? match[2] : '';
+      const loadAddress = match && match.length > 3 ? match[3] : '';
+      const file = match && match.length > 6 ? match[6] : '';
+      const line = match && match.length > 7 ? match[7] : '';
+
+      // TODO: Why doesn't this match the BacktraceFrame type?
```

Comment:
We don't support a `stack_adddress` property as far as I know. Can someone write up a proposal if that's something we should consider? Keep in mind that we try to keep the backtrace representation as generic as possible so that other languages can use the property if they have a similar feature.

Reply:
This I also copied directly from the original code. I was confused about `stack_address` and lack of `column` but happy to look further into it!

Reply:
I think it makes more sense to remove at this point, since we don't even know what it was supposed to be used for. Unless @andreybutov has some idea?

Reply:
Ok, so I spent some time looking deeper into this. I had tried to avoid needing to understand how the iOS stack traces work by just copying @andreybutov's original code, but alas I've now learned enough to know how much I don't know  😆 

The `localizedDescription` here looks something like: 
```
Exception 'Testing native iOS exception' was thrown while invoking throwErr on target ThrowErrModule with params (
)
callstack: (
	0   CoreFoundation                      0x00007ff8004288ab __exceptionPreprocess + 242
	1   libobjc.A.dylib                     0x00007ff80004dba3 objc_exception_throw + 48
	2   AwesomeProject                      0x00000001088b2ab0 -[AppDelegate application:didFinishLaunchingWithOptions:] + 0
	3   CoreFoundation                      0x00007ff80042f2fc __invoking___ + 140
	4   CoreFoundation                      0x00007ff80042c529 -[NSInvocation invoke] + 305
	5   CoreFoundation                      0x00007ff80042c7bc -[NSInvocation invokeWithTarget:] + 70
	6   AwesomeProject                      0x0000000108dd9aa6 -[RCTModuleMethod invokeWithBridge:module:arguments:] + 2534
	7   AwesomeProject                      0x0000000108dde69b _ZN8facebook5reactL11invokeInnerEP9RCTBridgeP13RCTModuleDatajRKN5folly7dynamicEiN12_GLOBAL__N_117SchedulingContextE + 2043
	8   AwesomeProject                      0x0000000108dddccc _ZZN8facebook5react15RCTNativeModule6invokeEjON5folly7dynamicEiENK3$_0clEv + 156
	9   AwesomeProject                      0x0000000108dddc29 ___ZN8facebook5react15RCTNativeModule6invokeEjON5folly7dynamicEi_block_invoke + 25
	10  libdispatch.dylib                   0x00007ff80013b7fb _dispatch_call_block_and_release + 12
	11  libdispatch.dylib                   0x00007ff80013ca3a _dispatch_client_callout + 8
	12  libdispatch.dylib                   0x00007ff800144346 _dispatch_lane_serial_drain + 949
	13  libdispatch.dylib                   0x00007ff800144fe3 _dispatch_lane_invoke + 400
	14  libdispatch.dylib                   0x00007ff800151b46 _dispatch_workloop_worker_thread + 870
	15  libsystem_pthread.dylib             0x00007ff836170fd0 _pthread_wqthread + 326
	16  libsystem_pthread.dylib             0x00007ff83616ff57 start_wqthread + 15
)
```
As far as I can tell, this call stack doesn't actually contain lines or columns, or even file names. Instead it has some kind of memory address. I found this helpful: https://www.bugsnag.com/blog/symbolicating-ios-crashes. I'm not sure if this has changed since Andrey worked on it, but the `file` and `line` variables in this code are undefined. 

Apparently in order to get something useful out of an error like this, we need to use a `dSYM` file. I'm not sure how to generate one automatically or how to use one, but I'm looking into it. 

I so far haven't found a good breakdown of exactly what each item in the call stack represents -- if anyone finds a good resource for this please lmk.

Reply:
@BethanyBerkowitz I have some notes on this. Let me get them in some sort of presentable order and I'll post them in Slack.

Reply:
> I'm not sure if this has changed since Andrey worked on it, but the file and line variables in this code are undefined.

If this is not something we introduced with this PR, my suggestion remains the same: let's tackle this with a separate issue, independently.

Reply:
https://github.com/honeybadger-io/honeybadger-js/issues/1082